### ISSUE TITLE
Default paper format

### ISSUE THREAD

**Issue by j1schmid** [Ordissimo/scangearmp2#24]
So far the default paperformat is selected according to the variable LANG. Wouldn't it be more reasonable to use the variable LC_PAPER. This would lead to the patch:

```
diff --git a/scangearmp2/src/mainui.c b/scangearmp2/src/mainui.c
index 31d50cf..b08d81b 100755
--- a/scangearmp2/src/mainui.c
+++ b/scangearmp2/src/mainui.c
@@ -106,7 +106,7 @@ static int ui_main_is_en_US( void )
 #ifdef USE_PO_LOCALE
        env_locale = _( "Locale" );
 #else
-       if( ( env_locale = (char *)getenv( "LANG" ) ) == NULL ){
+       if( ( env_locale = (char *)getenv( "LC_PAPER" ) ) == NULL ){
                env_locale = "";
        }
        env_locale = (char *)strsep( &env_locale, "." );
```

**Comment by Ordissimo** [comment]
Hi, 
I had little time, I just looked at your patch.
My Git concerns the support of canon drivers by sane.
So I made some modifications.
Support for different resolutions (this is not the case for scangearmp2, only 300 resolution is used).
Possibility to adjust the scan size according to the device used.
Added ADF support.

**Comment by Ordissimo** [comment]
If you offer me a "pull-request", I would merge it!

**Comment by j1schmid** [comment]
I do not understand all of the comment:
> Hi,
I had little time, I just looked at your patch.
My Git concerns the support of canon drivers by sane.
So I made some modifications.
Support for different resolutions (this is not the case for scangearmp2, only 300 resolution is used).
Possibility to adjust the scan size according to the device used.
Added ADF support.

Is there possibly something mixed up with pull request #31 ?

However. For sure I can offer a pull request, but so far I do not have the permissions to create a new branch or push something.

**Comment by Ordissimo** [comment]
I have completed the features offered by the canon_pixma backend.
Among these features :
- Choice of resolutions supported by the device. Scangearmp2 doesn't offer a choice of resolution.
- Choice of the source, the glass or the tray, depending on the device.
- Possibility to choose the paper size and the possibility to crop the scan.
To test these features use xsane.
I will merge the "pull-request" after testing. After the holidays...

To propose a "pull-request", you have to clone the project.
Apply your modification and propose it to me.
The web interface allows these actions.

**Comment by j1schmid** [comment]
Thanks for your advice regarding "pull-request".

I created the "pull-request" #34 where the default paper size is chosen according to the locale environment variable LC_PAPER.

**Comment by Ordissimo** [comment]
Fixed by #34
